v1/plugin/plugin.go

```diff
@@ -101,16 +146,48 @@ type tlsServerSetup interface {
 }
 
 // osInputOutput supports interactions with OS for the plugin lib
-type osInputOutput interface {
+type OSInputOutput interface {
 	// readOSArgs gets command line arguments passed to application
 	readOSArgs() []string
 	// printOut outputs given data to application standard output
 	printOut(data string)
+
+	args() int
+
+	setContext(c *cli.Context)
 }
 
 // standardInputOutput delivers standard implementation for OS
 // interactions
 type standardInputOutput struct {
+	context *cli.Context
+}
+
+// libInputOutput holds utility used for OS interactions
+var libInputOutput OSInputOutput = &standardInputOutput{}
+
+// readOSArgs implementation that returns application args passed by OS
+func (io *standardInputOutput) readOSArgs() []string {
+	if io.context != nil {
+		return io.context.Args().Tail()
```

Comment:
This should use First() not Tail.  Tail excludes the first argument.  I'll submit a commit that fixes. :)